server/controllers/register.js

```diff
@@ -0,0 +1,31 @@
+const bcrypt = require('bcrypt');
+
+
+const {checkEmail, addUser} = require('../database/queries');
+const {regSchema} = require('../utils/validations')
+
+const signup = (req, res) => {
+    const user = req.body;
+
+    const userValid = regSchema.validateAsync(user);
+
+    const userExist = (rows) => {
+        if(rows.rowCount > 0){
+            throw new Error('user registered');
+        }
+
+    };
+
+    userValid
+    .then((user) => checkEmail(user.email))
+    .then(rows => userExist(rows))
```

Comment:
```suggestion
    .then({ rowCount } => userExist(rowCount))
```

Reply:
why did you change that?

Reply:
Because you are only using `rowCount` in the `userExist` function, so why we should send the whole object?

Reply:
```js
    const userExist = (rowCount) => {
        if(rowCount > 0){
            throw new Error('user registered');
        }

    };
```